src/language-yaml/printer-yaml.js

```diff
@@ -113,7 +113,8 @@ function genericPrint(path, options, print) {
       lineSuffix([
         node.type === "mappingValue" && !node.content ? "" : " ",
         path.parent.type === "mappingKey" &&
-        path.getParentNode(2).type === "mapping" &&
+        (path.getParentNode(2).type === "mapping" ||
+          path.getParentNode(2).type === "flowMapping") &&
```

Comment:
Can you rewrite this with `path.match`? Or use

`["mapping", "flowMapping"].includes(path.getParentNode(2).type)`, so we don't need visit parent twice.